dev-qt/qt3d/files/qt3d-5.15.2-riscv-atomic.patch

```diff
@@ -0,0 +1,10 @@
+diff --git a/src/plugins/renderers/opengl/opengl.pri b/src/plugins/renderers/opengl/opengl.pri
+index 1682ab7..e0ca986 100644
+--- a/src/plugins/renderers/opengl/opengl.pri
++++ b/src/plugins/renderers/opengl/opengl.pri
+@@ -39,3 +39,5 @@ qtConfig(qt3d-simd-sse2):!qtConfig(qt3d-simd-avx2) {
+     CONFIG += simd
+     QMAKE_CXXFLAGS += $$QMAKE_CFLAGS_SSE2
+ }
++
++gcc:isEqual(QT_ARCH, "riscv64"):qtConfig(std-atomic64): LIBS += -latomic
```

Comment:
It seems wrong @dlan17 
qtConfig(std-atomic64) has 64bit atomic. But it should determine !qtConfig(std-atomic64) to use latomic.
Original error is `__atomic_exchange_1` missing. ATOMIC_NON64BIT is needed

Reply:
https://doc.qt.io/qt-5/qmake-test-function-reference.html#qtcompiletest-test

qtcompiletest can resolve it

Reply:
In use qtcompiletest case, it needs two test-qt-projects in patch. It looks not graceful

Reply:
New patch solve it.